pyiron_base/generic/hdfio.py

```diff
@@ -26,58 +25,13 @@
 __status__ = "production"
 __date__ = "Sep 1, 2017"
 
-
-class HDFStoreIO(pandas.HDFStore):
-    """
-    dict-like IO interface for storing pandas objects in PyTables either Fixed or Table format.
-    - copied from pandas.HDFStore
-
-    Args:
-        path (str) : File path to HDF5 file
-        mode (str): {'a', 'w', 'r', 'r+'}, default 'a'
-                    ``'r'``
-                        Read-only; no data can be modified.
-                    ``'w'``
-                        Write; a new file is created (an existing file with the same name would be deleted).
-                    ``'a'``
-                        Append; an existing file is opened for reading and writing, and if the file does not exist it
-                        is created.
-                    ``'r+'``
-                        It is similar to ``'a'``, but the file must already exist.
-        complevel (int): 1-9, default 0
-                         If a complib is specified compression will be applied
-                         where possible
-        complib (str): {'zlib', 'bzip2', 'lzo', 'blosc', None}, default None
-                       If complevel is > 0 apply compression to objects written in the store wherever possible
-        fletcher32 (bool): bool, default False
-                           If applying compression use the fletcher32 checksum
-    """
-
-    def __init__(
-        self, path, mode=None, complevel=None, complib=None, fletcher32=False, **kwargs
-    ):
-        super(HDFStoreIO, self).__init__(
-            path,
-            mode=mode,
-            complevel=complevel,
-            complib=complib,
-            fletcher32=fletcher32,
-            **kwargs
-        )
-
-    def open(self, **kwargs):
-        """
-        Open the file in the specified mode - copied from pandas.HDFStore.open()
-
-        Args:
-            **kwargs: mode : {'a', 'w', 'r', 'r+'}, default 'a'
-                      See HDFStore docstring or tables.open_file for info about modes
-        Returns:
-            HDFStoreIO: self - in contrast to the original implementation in pandas.
-        """
-        super(HDFStoreIO, self).open(**kwargs)
-        return self
-
+def open_hdf5(filename, mode="r", swmr=False):
+    if swmr and mode != "r":
+        store = h5py.File(filename, mode=mode, libver="latest")
+        store.swmr = True
```

Comment:
what is the difference between 
`h5py.File(..., swmr=True)` and `store = h5py.File(...); store.swmr = True`?
Since if this is the same, all the logic is not needed?

Reply:
When opening the file writing/amending h5py ignores the `swmr` flag as in #298, this has somehow something to do with what you are allowed to do with a file [swmr mode](https://docs.h5py.org/en/stable/swmr.html).  It's a bit academic, because we never use `swmr=True`, so I could throw it out.  It's only in, because I started working on this before #298 and only rebased after.